cities/crud.py

```
@@ -0,0 +1,173 @@
+    await db.commit()
+
+    updated_city_query = select(CityDB).where(CityDB.id == city_id)
+    updated_city_result = await db.execute(updated_city_query)
+    return updated_city_result.scalar()
+
+
+async def city_delete(db: AsyncSession, city: CityDB) -> str:
+    success_message = f"City: '{city.name}' was successfully deleted"
+    await db.delete(city)
+    await db.commit()
+    return success_message
+
+
+async def get_cities(db: AsyncSession) -> list[City]:
+    query = select(CityDB)
+    result = await db.execute(query)
+    cities = result.scalars().all()
+    return [City.model_validate(city) for city in cities]
+
+
+async def temperatures_create(db: AsyncSession, new_data: list[dict]):
+    try:
+        logger.info(
+            "Starting to insert temperatures for %d cities",
+            len(new_data)
+        )
+        query = insert(models.TemperatureDB).values(new_data)
+        await db.execute(query)
+        await db.commit()
+        logger.info(
+            "Successfully inserted temperatures for %d cities",
+            len(new_data)
+        )
+        return f"Temperatures for {len(new_data)} cities have been updated."
+    except SQLAlchemyError as e:
+        await db.rollback()
+        logger.error(
+            "An error occurred while inserting temperatures: %s",
+            str(e)
+        )
+        return f"An error occurred: {str(e)}"
+
+
+async def get_temperatures(db: AsyncSession) -> list[Temperature]:
+    try:
+        logger.info("Fetching all temperatures from the database.")
+        query = select(TemperatureDB).options(joinedload(TemperatureDB.city))
+        result = await db.execute(query)
+        temperatures = result.scalars().all()
+
+        logger.info(
+            "Successfully retrieved %d temperatures.",
+            len(temperatures)
+        )
+
+        return [
+            Temperature.model_validate(temperature)
+            for temperature in temperatures
+        ]
+
+    except Exception as e:
```

Catching a general `Exception` is not recommended as it can mask other unexpected errors. Consider catching more specific exceptions related to the database operations.